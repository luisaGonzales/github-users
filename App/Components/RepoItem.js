import React, {Component} from 'react';
import {Text, Right, ListItem, Body } from 'native-base';
import styles from '../Styles/ListItemRepoStyle';

class RepoItem extends Component {
  constructor(props) {
    super(props);
  }

  render(){
    return(
      <ListItem 
        avatar
        style={styles.listItemRepoContainer }>
        <Body style={styles.noneBorder}>
          <Text>
            {this.props.name}
          </Text>
          <Text style={styles.subtitle} note>{this.props.description}</Text>
        </Body>
        <Right style={styles.noneBorder}>
          <Text style={styles.subtitle}>PR Count: {this.props.pullRequests}</Text>
        </Right>
      </ListItem>
    )
  }
}

export default RepoItem;
