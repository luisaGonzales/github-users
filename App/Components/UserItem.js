import React, { Component } from 'react';
import styles from '../Styles/ListItemStyles';
import {Text, List, Left, ListItem, Thumbnail, Body, Content, View} from 'native-base';

class UserItem extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <ListItem avatar
          onPress={() => this.props.navigation.navigate('Details', {
            login: this.props.login
          })}
          style={styles.listItemContainer}
        >
          <Left>
            {
              this.props.avatar ?
              <Thumbnail source={{ uri: this.props.avatar }} />
              :
              <Thumbnail source={require('../Images/github-icon.png')} />
            }
          </Left>
          <Body style={styles.listItemBody}>
            <Text>
              {this.props.name}, {this.props.location}
            </Text>
            <Text note>{this.props.login}</Text>
          </Body>
      </ListItem>
    )
  }
}

export default UserItem;