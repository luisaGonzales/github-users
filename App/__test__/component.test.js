import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UserItem from '../Components/UserItem';
import RepoItem from '../Components/RepoItem';
import GET_USERS from '../Querys/getUsers';
import GET_REPOS from '../Querys/getRepos';

//Testing render items
test('Render user item with null data', () => {
  const userItemNull = renderer.create(
    <UserItem 
      login={ "No user login" } 
      avatar={ null } 
      name={ "No name" } 
      location={ "No location" }
    />
  ).toJSON();
  expect(userItemNull).toMatchSnapshot();
});

test('Render user item with data', () => {
  const userItem = renderer.create(
    <UserItem 
      login={ "luisaGonzales" } 
      avatar={ "https://avatars3.githubusercontent.com/u/29384758?v=4" } 
      name={ "Luisa Gonzales" } 
      location={ null }
    />
  ).toJSON();
  expect(userItem).toMatchSnapshot();
});

test('Render repo item', () => {
  const repoItem = renderer.create(
    <RepoItem 
      name={"IA-tictactoe"}
      description={"N en raya con algoritmo de MinMax"}
      pullRequests={0}
    />
  ).toJSON();
  expect(repoItem).toMatchSnapshot();
});

//Testing data querys
test('Get users query without errors', () => {
 expect(GET_USERS).toMatchSnapshot();
});

test('Get repos query without errors', () => {
  expect(GET_REPOS).toMatchSnapshot();
});

//Testing item properties
test('Check properties ', () => {
  expect(UserItem).toHaveProperty("name");
});








