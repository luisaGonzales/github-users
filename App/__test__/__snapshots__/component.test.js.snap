// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get repos query without errors 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "GetRepositories",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "login",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "login",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "user",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "first",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "count",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "after",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "cursor",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "repositories",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "edges",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "cursor",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "directives": Array [],
                              "kind": "FragmentSpread",
                              "name": Object {
                                "kind": "Name",
                                "value": "RepoFragment",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "login",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "cursor",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Int",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "count",
            },
          },
        },
      ],
    },
    Object {
      "directives": Array [],
      "kind": "FragmentDefinition",
      "name": Object {
        "kind": "Name",
        "value": "RepoFragment",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "node",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "description",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "pullRequests",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "totalCount",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "typeCondition": Object {
        "kind": "NamedType",
        "name": Object {
          "kind": "Name",
          "value": "RepositoryEdge",
        },
      },
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 382,
    "start": 0,
  },
}
`;

exports[`Get users query without errors 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "SearchUsers",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "query",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "user",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "type",
                },
                "value": Object {
                  "kind": "EnumValue",
                  "value": "USER",
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "first",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "counter",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "after",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "cursor",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "search",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "edges",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "cursor",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "node",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "directives": Array [],
                              "kind": "FragmentSpread",
                              "name": Object {
                                "kind": "Name",
                                "value": "UserFragment",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "user",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "cursor",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Int",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "counter",
            },
          },
        },
      ],
    },
    Object {
      "directives": Array [],
      "kind": "FragmentDefinition",
      "name": Object {
        "kind": "Name",
        "value": "UserFragment",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "name",
            },
            "selectionSet": undefined,
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "location",
            },
            "selectionSet": undefined,
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "login",
            },
            "selectionSet": undefined,
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "avatarUrl",
            },
            "selectionSet": undefined,
          },
        ],
      },
      "typeCondition": Object {
        "kind": "NamedType",
        "name": Object {
          "kind": "Name",
          "value": "User",
        },
      },
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 329,
    "start": 0,
  },
}
`;

exports[`Render repo item 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    avatar={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": null,
          "paddingRight": null,
          "paddingTop": null,
        },
        Object {
          "borderBottomColor": "#ececec",
          "borderBottomWidth": 1,
          "marginLeft": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          },
          Object {
            "borderBottomColor": "transparent",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        uppercase={false}
      >
        IA-tictactoe
      </Text>
      <Text
        note={true}
        style={
          Array [
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": null,
              "marginRight": 10,
            },
            Object {
              "color": "#b8b8b8",
              "fontSize": 12.8,
            },
          ]
        }
        uppercase={false}
      >
        N en raya con algoritmo de MinMax
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "alignSelf": "stretch",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 0,
            "paddingBottom": 10,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": 10,
          },
          Object {
            "borderBottomColor": "transparent",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": null,
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#b8b8b8",
              "fontSize": 12.8,
            },
          ]
        }
        uppercase={false}
      >
        PR Count: 
        0
      </Text>
    </View>
  </View>
</View>
`;

exports[`Render user item with data 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    avatar={true}
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": null,
          "paddingRight": null,
          "paddingTop": null,
        },
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#ececec",
          "marginLeft": -16,
          "paddingBottom": 16,
          "paddingLeft": 32,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "flex-start",
          "flex": 0,
          "flexDirection": "row",
          "paddingTop": 14,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://avatars3.githubusercontent.com/u/29384758?v=4",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "width": 56,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          },
          Object {
            "borderBottomColor": "transparent",
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        uppercase={false}
      >
        Luisa Gonzales
        , 
      </Text>
      <Text
        note={true}
        style={
          Object {
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "200",
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        uppercase={false}
      >
        luisaGonzales
      </Text>
    </View>
  </View>
</View>
`;

exports[`Render user item with null data 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    avatar={true}
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": null,
          "paddingRight": null,
          "paddingTop": null,
        },
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#ececec",
          "marginLeft": -16,
          "paddingBottom": 16,
          "paddingLeft": 32,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "flex-start",
          "flex": 0,
          "flexDirection": "row",
          "paddingTop": 14,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../App/Images/github-icon.png",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "width": 56,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": null,
            "alignSelf": null,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flex": 1,
            "marginLeft": 15,
            "paddingBottom": 10,
            "paddingTop": 10,
          },
          Object {
            "borderBottomColor": "transparent",
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        uppercase={false}
      >
        No name
        , 
        No location
      </Text>
      <Text
        note={true}
        style={
          Object {
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "200",
            "marginLeft": null,
            "marginRight": 10,
          }
        }
        uppercase={false}
      >
        No user login
      </Text>
    </View>
  </View>
</View>
`;
